File: javascript/rna-transcription/rna-transcription.js
//
// This is only a SKELETON file for the 'RNA Transcription' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const toRna = (nucleotide) => {
  if ( nucleotide.includes("X") ) {
    // throw new Error if above evals to true
    throw new Error('Invalid input');
    // else transcribe DNA to RNA
  } else {
    // replacement object
    const replacements = {'A': 'U', 'T': 'A', 'G': 'C', 'C': 'G'};
    // replace() method returns a new string with some/all matches of a pattern
    // replaced by a replacement.
    return nucleotide.replace(/A|T|G|C/g, char => replacements[char]);
  }
};

File: ruby/rna-transcription/rna_transcription.rb
=begin
Write your code for the 'Rna Transcription' exercise in this file. Make the tests in
`rna_transcription_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/rna-transcription` directory.
=end

class Complement
  def self.of_dna dna
    # check if dna includes invalid input
    if dna =~ /[UX]/
      ''
    else
      # gsub returns a copy of dna with all the occurrences
      # of the pattern (regex) substituted for the second argument
      dna.gsub(/[GCTA]/, 'G' => 'C', 'C' => 'G', 'T' => 'A', 'A' => 'U')
    end
  end
end
